app/views/ReportingFrequencyView.scala.html

```diff
@@ -31,7 +31,17 @@
     govukInsetText : GovukInsetText
 )
 
-@(viewModel: ReportingFrequencyViewModel)(implicit messages: Messages, user: auth.MtdItUser[_])
+@(viewModel: ReportingFrequencyViewModel)(implicit messages: Messages, user: auth.MtdItUser[_], dateService: DateServiceInterface)
+
+@optInLink = {
```

Comment:
If we can avoid usage of .get over Option and .head over List; you should be able to move this behaviour on the view model level;